File: app/src/main/java/com/example/myapplication/PhotoFragment.java
package com.example.myapplication;

import androidx.lifecycle.ViewModelProvider;

import android.content.pm.ActivityInfo;
import android.graphics.Bitmap;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import org.opencv.core.Mat;
import org.opencv.android.Utils;

public class PhotoFragment extends Fragment {

    private ScannerViewModel mViewModel;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {

        mViewModel = new ViewModelProvider(requireActivity()).get(ScannerViewModel.class);
        View view = inflater.inflate(R.layout.photo_fragment, container, false);

        Mat snapShot = mViewModel.getFrame();
        Bitmap bitMap = Bitmap.createBitmap(snapShot.cols(), snapShot.rows(),Bitmap.Config.ARGB_8888);
        Utils.matToBitmap(snapShot, bitMap);
        ImageView imgViewBefore = (ImageView) view.findViewById(R.id.imgViewBefore);
        imgViewBefore.setImageBitmap(bitMap);

        Mat processedMat = mViewModel.makeScanning(snapShot, true);
        bitMap = Bitmap.createBitmap(processedMat.cols(), processedMat.rows(),Bitmap.Config.ARGB_8888);
        Utils.matToBitmap(processedMat, bitMap);
        ImageView imgViewAfter = (ImageView) view.findViewById(R.id.imgViewAfter);
        imgViewAfter.setImageBitmap(bitMap);

        //getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);

        return view;
    }

}

File: app/src/main/java/com/example/myapplication/PhotoFragment.kt
package com.example.myapplication

import android.content.Intent
import android.graphics.Bitmap
import android.net.Uri
import android.os.Bundle
import android.provider.MediaStore
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.example.myapplication.databinding.PhotoFragmentBinding
import org.opencv.core.Core
import org.opencv.android.Utils as OpenCVUtils


class PhotoFragment : Fragment() {
    private lateinit var mViewModel: ScannerViewModel
    private lateinit var binding: PhotoFragmentBinding
    private lateinit var mImagePath: String
        override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        mViewModel = ViewModelProvider(requireActivity()).get(ScannerViewModel::class.java)
        binding = PhotoFragmentBinding.inflate(layoutInflater, container, false)
        mViewModel.liveData().observe(viewLifecycleOwner, Observer { snapShot ->
            Log.d(Utils.TAG, " frame processing with number of cols: " + snapShot.cols() + " rows: " + snapShot.rows())
            if (snapShot.cols() > 0 && snapShot.rows() > 0) {
                var bitMap = Bitmap.createBitmap(snapShot.cols(), snapShot.rows(), Bitmap.Config.ARGB_8888)
                OpenCVUtils.matToBitmap(snapShot, bitMap)
                val imgViewBefore = binding.imgViewBefore
                imgViewBefore.setImageBitmap(bitMap)
                Utils.makeScanning(snapShot, false)?.let {
                    bitMap =
                        Bitmap.createBitmap(it.cols(), it.rows(), Bitmap.Config.ARGB_8888)
                    OpenCVUtils.matToBitmap(it, bitMap)
                    val imgViewAfter = binding.imgViewAfter
                    imgViewAfter.setImageBitmap(bitMap)
                    mImagePath = MediaStore.Images.Media.insertImage(
                        requireActivity().contentResolver,
                        bitMap,
                        "ScannedImage",
                        null
                    )
                }
            } else {
                Log.e(Utils.TAG, " invalid frame processed: ")
            }
        })

        binding.saveButton.setOnClickListener {
            val emailIntent = Intent(Intent.ACTION_SEND)
            emailIntent.type = "application/image"
            emailIntent.putExtra(Intent.EXTRA_EMAIL, arrayOf("eamalafeev@gmail.com"))
            emailIntent.putExtra(Intent.EXTRA_SUBJECT, "Scanned img")
            emailIntent.putExtra(Intent.EXTRA_TEXT, "From Scanner app")
            val uri: Uri = Uri.parse(mImagePath)
            emailIntent.putExtra(Intent.EXTRA_STREAM, uri)
            startActivity(Intent.createChooser(emailIntent, "Send mail..."))
        }

        return binding.root
    }
}